### ISSUE TITLE
Omni-bot on macOS

### ISSUE THREAD

**Issue by plastovicka** [etlegacy/etlegacy#2566]
Omni-bot for Mac was released in January 2024. You can enable option **FEATURE_OMNIBOT** in CMakeLists.txt and easybuild.sh.

**Comment by jackeri** [comment]
I know, I made the pr to the GitHub repository for the initial macOS support, I just haven't enabled the build by default yet. I figured it's not "officially" supported yet as the last stable omnibot was from last year, which did not have the support yet included. But yeah, it's on my to-do for the next release of legacy.

**Comment by plastovicka** [comment]
macOS is officially supported in Omni-bot 0.91

**Comment by plastovicka** [comment]
Can you remove messages "No Omni-Bot server support!" from the [download](https://www.etlegacy.com/download) page ?

**Comment by Aciz** [comment]
We currently do not ship omni-bot on macOS at all, because it wasn't supported earlier, and a compatible version was never added to the package after the upstream added support for it. We should probably start shipping that at some point @jackeri?

In the meanwhile, I added a note & link to the download page about 0.91 having support for it.

![Image](https://github.com/user-attachments/assets/d4ff841b-2b48-44c7-8ac5-d8a2192c62d2)

**Comment by ZehuaZhang** [comment]
where do I put the omni-bot folder, I put into the legacy folder, not working? Do I put the subfolder in where?

**Comment by rafal1137** [comment]
@ZehuaZhang Put `omni-bot` into legacy at installation directory and make sure you have correct path and you have enabled them
```
omnibot_path = "./legacy/omni-bot"
omnibot_enable = "1"
```

**Comment by jackeri** [comment]
I don't remember if the previous release was compiled with OmniBot support or not for sure. But the next one will for sure support it.

**Comment by ZehuaZhang** [comment]
Thanks for the replies, but it doesn't work, what did I do wrong?

`legacy` Folder
<img width="183" height="260" alt="Image" src="https://github.com/user-attachments/assets/7ad10eb4-9aa1-4678-b05a-f8e56b1d3901" />

`etmain\legacy.cfg`
```
// OMNI-BOT

set omnibot_enable "1"
set omnibot_path "./legacy/omni-bot"
set omnibot_flags "0"
```

In game console

<img width="804" height="698" alt="Image" src="https://github.com/user-attachments/assets/33f77b1e-eced-4ba5-8a2d-e9f289c4d00b" />

**Comment by rafal1137** [comment]
@ZehuaZhang Welll this is purelly client console. There is no logs about omni-bot loading or failing to load so either no support for current release you downloaded as @jackeri mentioned. Also latest snapshots does not provide you with omni-bot files too so yeah
no bots for you untill next release or you can download latest snapshot from https://www.etlegacy.com/workflow-files/ and check.

**Comment by mittermichal** [comment]
> Also latest snapshots does not provide you with omni-bot files too so yeah

you can download omni-bot separatelly from https://github.com/jswigart/omni-bot/releases

look for omnibot initialization log lines:

<img width="948" height="488" alt="Image" src="https://github.com/user-attachments/assets/98631055-0c57-42dc-849d-75249828a9cc" />

omni-bot initialization will be atetmpted if you have `omnibot_enabled 1`
you can check and change that cvar in console

**Comment by ZehuaZhang** [comment]
I hosted a game, I browsed all the previous logs in console, there's no log lines for `omni-bot`. The previous image is from a server I created, I tried to set in the host console, nothing happening.

I looked at some youtube videos, the server console should indicate omni-bot successfully initialized.

**Comment by mittermichal** [comment]
> I browsed all the previous logs in console, there's no log lines for omni-bot. The previous image is from a server I created

then you don't have `omnibot_enabled 1` set

**Comment by Aciz** [comment]
Please post the full console output, your screenshot does not show the part of the initialization process where omni-bot would get loaded. Start the server, type `/condump log.txt` and post it here. It will be saved to `legacy` directory in your homepath.

**Comment by plastovicka** [comment]
You have to download file **omni-bot_0_92_ET_waypoint_mod.zip**. Extract omnibot folder to ETLegacy. Then start the game and select omnibot mod from **MODS** menu.

**Comment by mittermichal** [comment]
> You have to download file **omni-bot_0_92_ET_waypoint_mod.zip**. Extract omnibot folder to ETLegacy. Then start the game and select omnibot mod from **MODS** menu.

I think @ZehuaZhang wants to play with the bots, not modify waypoints, etc.

**Comment by plastovicka** [comment]
You can use the omnibot waypoint mod to play with the bots. But it is original ET without new legacy features (MP34 weapon etc.).

**Comment by mittermichal** [comment]
yes, but mod is not the problem here if I understand correctly, its the misconfiguration.

**Comment by ZehuaZhang** [comment]
This is quite misleading because the download page makes it appear that the legacy mod works together with omni-bot rather than being a separate mod. 

I downloaded the Windows version, and omni-bot is included in legacy folder. As I remembered correctly in the past, I could create bots in legacy. The omni-bot README also states this.

```
- Install MODs
Omni-Bot does not work with vanilla etmain. You need some mod. You can use our omnibot mod or you can choose some 3rd party mods. It's possible to have multiple mods installed. There are many mods that support Omni-bot: Bastardmod, ETBlight, ETNam, ETPub, infected, Jaymod, legacy, N!tmod, NoQuarter, silEnT.


- Install Omni-Bot
Note: If you have ET:Legacy, you will find Omni-Bot already installed in the legacy folder.
There is omni-bot folder inside the ZIP archive. Extract it. Omni-Bot can be installed to any location. It is not necessary to have omni-bot inside Enemy Territory folder. It is recommended to install it into folder which has permission to write files. 
```

**Comment by mittermichal** [comment]
are you on Windows or macOS?

**Comment by plastovicka** [comment]
I think the problem is in CMakeLists.txt:
`cmake_dependent_option(FEATURE_OMNIBOT	"Enable Omni-bot support (mod)"	ON "( NOT MINGW AND WIN32 ) OR FORCE_OMNIBOT OR CMAKE_SYSTEM_NAME MATCHES Linux" OFF)
`
Can anybody fix it and rebuild workflow-files snapshot binaries ?

**Comment by jackeri** [comment]
Yeah, I'll finally enable it after I've gotten the sdl3 port done.
